rack-protection/lib/rack/protection/base.rb

```diff
@@ -93,7 +93,10 @@ def session(env)
       end
 
       def drop_session(env)
-        session(env).clear if session? env
+        return unless session? env
+
+        warn env, "session dropped by #{self.class}"
```

Comment:
I think we need to make it easy to silence this warning. Maybe via an `ENV`?

WDYT @jkowens ?

Reply:
That sounds like a really good idea 👍

Reply:
@jdelStrother any interest in continuing here?

Reply:
I don't really get why dropping sessions might be a noisy warning that needs silencing, but `deny` and `report` warnings are fine and should always be printed. Could you clarify?

If we're adding an ENV var, any thoughts on naming? `RACK_PROTECTION_WARN_ON_DROPPED_SESSIONS=0` ?

Reply:
because it is the default in Sinatra

We can warn by default but there should be an easy way to get back the previous behavior (no warning)

So it should be named "silence ..." something. The RACK_PROTECTION prefix is a good call 👍🏻

Reply:
You don't think it should just silence all warnings, along the lines of this ?

```diff
diff --git a/rack-protection/lib/rack/protection/base.rb b/rack-protection/lib/rack/protection/base.rb
index 6f7e81fb..1ffc37c6 100644
--- a/rack-protection/lib/rack/protection/base.rb
+++ b/rack-protection/lib/rack/protection/base.rb
@@ -10,7 +10,7 @@ module Rack
   module Protection
     class Base
       DEFAULT_OPTIONS = {
-        reaction: :default_reaction, logging: true,
+        reaction: :default_reaction, logging: ENV["RACK_PROTECTION_SILENCE_REACTIONS"] != "1",
         message: 'Forbidden', encryptor: Digest::SHA1,
         session_key: 'rack.session', status: 403,
         allow_empty_referrer: true,

```

Reply:
No, I think it should be specific to drop session

Reply:
OK, see what you think to the latest push.